DOCUMENTATION.md

```diff
@@ -0,0 +1,101 @@
+
+If you also want to run and serve the M3 documentation, you need the custom theme, [Victor](https://github.com/chronosphereio/victor). Clone or download it into the same directory as the M3 codebase.
+
+To run some tests and test the production build steps, you need [Docker](https://www.docker.com).
+
+## Folder Structure
+
+Hugo hosts the M3 documentation, website, and Open API spec.
+
+### Website
+
+The website is the HTML, CSS, and JavaScript files in the _site/static_ folder. Edit those files to make any changes.
+
+### Open API Docs
+
+Hugo hosts the Open API spec from the _site/static/openapi_ folder. To make changes to the spec, edit the _spec.yml_ file in _src/query/generated/assets_, and a CI job generates the file that Hugo hosts automatically.
+
+### Configuration
+
+The _site/config_ folder has configuration for Hugo, split into three folders:
+
+-   A default  _config.toml_ file in _site/config/\_default_
+-   Overridden configuration for development and production environments in _site/config/development_ and _site/config/production_ respectively.
+
+### Theme Overrides
+
+The _site/layouts_ folder adds a several changes and additional files to the Victor theme used by the M3 documentation.
+
+### Documentation
+
+The _site/content_ folder contains the documentation files, organized by folders that match the paths for URLs. The _includes_ folder is a special folder not served as part of the documentation and contains files used by other files.
+
+#### Theme
+
+The M3 documentation uses its own (open source) theme, Victor. You can read all the features that theme provides in [the repository for Victor](https://github.com/chronosphereio/victor).
+
+Victor is a theme based on Hugo modules, [read more in the Hugo docs](https://gohugo.io/hugo-modules/use-modules/) about how to use and update it.
+
+#### Shortcodes
+
+The M3 documentation adds the following additional shortcodes:
+
+-   `{{% apiendpoint %}}` - Combines the values of `Params.api.localCordinator` + `Params.api.apiEndpoint` as defined in the site configuration to output the base API endpoint for M3 running on a localhost.
+-   `{{% docker-version %}}` - Outputs the value of `Params.releases.docker` as defined in the site configuration to output the consistent current Docker release.
+-   `{{% now %}}` - Outputs the current Unix timestamp to allow for up to date timestamps in code examples.
+
+## Running Documentation
+
+As noted in the prerequisites section, if you want to run the documentation locally with `hugo server`, you need to have the Victor theme in the same directory as the M3 codebase, as `hugo server` runs in Hugo's "development" mode (and matches _site/config/development/config.toml_).
+
+This does mean that as you make changes to the theme or documentation, usually, it refreshes automatically in the browser preview. Sometimes Hugo doesn't refresh included files, so you may need to restart the server process.
```

Comment:
Do we want to add this? I feel like involving the themeing can get confusing.. Not sure how critical it is for people to preview their edits / contributions?

Reply:
@gibbscullen To some it may be important, but I'll clarify this is only strictly necessary if you want to run it locally. To most documentation type people, they also won't consider this particularly unusual.